src/main/java/io/github/cottonmc/cotton/gui/widget/WButton.java

```diff
@@ -99,14 +101,14 @@ public void paint(MatrixStack matrices, int x, int y, int mouseX, int mouseY) {
 		float buttonWidth = halfWidth*px;
 		float buttonHeight = 20*px;
 		
-		float buttonEndLeft = (200-(getWidth()/2)) * px;
+		float buttonEndLeft = (200-((float)getWidth()/2)) * px;
 
 		Identifier texture = getTexture();
 		ScreenDrawing.texturedRect(matrices, x, y, getWidth()/2, 20, texture, buttonLeft, buttonTop, buttonLeft+buttonWidth, buttonTop+buttonHeight, 0xFFFFFFFF);
 		ScreenDrawing.texturedRect(matrices, x+(getWidth()/2), y, getWidth()/2, 20, texture, buttonEndLeft, buttonTop, 200*px, buttonTop+buttonHeight, 0xFFFFFFFF);
 
 		if (icon != null) {
-			icon.paint(matrices, x + 1, y + 1, 16);
+			icon.paint(matrices, x+2, y+(BUTTON_HEIGHT-iconSize)/2, iconSize);
```

Comment:
Mind extracting the constant 2 here to a field like `ICON_OFFSET`? It's referenced in multiple places (like the label rendering below) so might be good to clarify.